airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/AutoDisableConnectionActivityImpl.java

```diff
@@ -73,13 +75,16 @@ public AutoDisableConnectionOutput autoDisableFailingConnection(final AutoDisabl
           }
         }
 
+        final boolean warningPreviouslySentForMaxDays =
+            numFailures > 1 && warningPreviouslySentForMaxDays(successTimestamp, maxDaysOfOnlyFailedJobsBeforeWarning, firstJob, jobs);
```

Comment:
Why do we check `numFailures > 1` here?

Reply:
We only send warnings/check for auto-disabling after a failure.

So if this is the first failure (`numFailures == 1`) then there would have been no opportunity to send a warning previously.

reading through this, makes sense to have that logic in `warningPreviouslySentForMaxDays` and i can just pass in `numFailures`

Reply:
Cool, that makes sense and yep, agreed that the logic sounds right in `warningPreviouslySentForMaxDays()`